Libraries/LibWeb/HTML/SessionHistoryTraversalQueue.cpp

```diff
@@ -31,32 +33,44 @@ SessionHistoryTraversalQueue::SessionHistoryTraversalQueue()
             m_timer->start();
             return;
         }
+
         while (m_queue.size() > 0) {
+            // Note: Check if current entry's load_document finished before executing the next entry
+            if (m_current_document && m_current_document->ready_state() == "loading" && m_current_document->is_active()) {
+                m_timer->start();
+                return;
+            }
```

Comment:
instead of making SHTQ responsible for checking the ready state, can we pass a Core::Promise into the "steps" and make SHTQ wait until this promise is resolved before proceeding to the next item? I think this change addresses one instance of a more general problem: although queued steps are considered synchronous, that’s not always true in practice. we need an API that allows us to explicitly signal when a step has completed.